hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/AbstractInternalQueryCache.java

```diff
@@ -185,7 +185,7 @@ protected void doFullValueScan(Predicate predicate, Set<V> resultingSet) {
 
             boolean valid = predicate.apply(queryEntry);
             if (valid) {
-                Object valueObject = queryEntry.getValue();
+                Object valueObject = toObject(queryEntry.getValueData());
```

Comment:
instead of eager cloning, can we make use of `com.hazelcast.spi.impl.UnmodifiableLazyList` as in [ClientMapProxy](https://github.com/hazelcast/hazelcast/blob/6a5bc11894e312366e82d4c808df31c2d441d0fc/hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java#L1355)?

Reply:
done